File: services/BillingService/Models/Invoice.cs
namespace BillingService.Models;

public class Invoice
{
    public int Id { get; set; }

    /// <summary>
    /// Numeração sequencial da nota, gerada por sequence do PostgreSQL.
    /// </summary>
    public int Number { get; set; }

    public InvoiceStatus Status { get; set; } = InvoiceStatus.Open;

    public DateTime CreatedAt { get; set; }

    public DateTime? ClosedAt { get; set; }

    public List<InvoiceItem> Items { get; set; } = [];
}

File: frontend/korp-web/src/app/models/invoice.ts
export type InvoiceStatus = 'Open' | 'Closed';

export interface InvoiceItem {
  id: number;
  productId: number;
  productCode: string;
  productDescription: string;
  quantity: number;
}

export interface Invoice {
  id: number;
  number: number;
  status: InvoiceStatus;
  createdAt: string;
  closedAt: string | null;
  items: InvoiceItem[];
}

export interface CreateInvoiceItem {
  productId: number;
  quantity: number;
}

export interface CreateInvoice {
  items: CreateInvoiceItem[];
}
